IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework/TestBuilder.cs

```diff
@@ -278,28 +280,31 @@ private object[] GatherTestBlockArguments(IServiceScope scope, MethodInfo execut
                 {
                     for (int i = 0; i < executeArgs.Length; i++)
                     {
-                        // Might be worth changing this to GetRequiredService and wrapping in a try/catch instead of checking if foundResult is null
-                        object foundResult =
+                        object foundResult = null;
+                        try
+                        {
+                            foundResult =
                             TestBlockResults.FirstOrDefault(tbr => tbr.GetType() == executeParams[i].ParameterType)
-                            ?? scope.ServiceProvider.GetService(executeParams[i].ParameterType);
-                        if (foundResult == null)
+                            ?? scope.ServiceProvider.GetRequiredService(executeParams[i].ParameterType);
+                        }
+                        catch(InvalidOperationException ex)
```

Comment:
Why this exception type?

Reply:
@Keboo That's the type that's thrown if GetRequiredService fails to find the asked for type. If this is too vague, I could use an exception filter to see if there are any details more specific to that scenario?

Reply:
Ah, I would say to just call GetService instead then and then you don't need to catch an exception.

Reply:
I was doing that, but I realized that there's not enough information to continue with the test at this point... so I _was_ doing that, but then immediately threw a new InvalidOperationException... this seemed like a more streamlined way to do it and keep the stacktrace informative. :)

Reply:
The only drawback to letting it throw and catch is that is extra overhead that is not needed. [GetRequiredService](https://source.dot.net/#Microsoft.Extensions.DependencyInjection.Abstractions/ServiceProviderServiceExtensions.cs,56) just does a null check on the return of GetService. I would propose that you do a similar null check, and simply assign TestBlockException = new InvalidOperationException("Useful stuff here").

Reply:
@Keboo Would that apply to the other instances of using GetRequiredService, too? I'd be hesitant to change one and add specific detailing here, but not other places (BUT I'm not opposed to changing. Just want to make sure I understand the purpose and apply it everywhere it's relevant. :) )

Reply:
I would say it should apply everywhere, you are relying on catching that exception. Exceptions are expensive to instantiate and handle, so if you can avoid trowing them and instead handle the case that is better.

Reply:
@Keboo Okay, changes pushed. I changed all cases of GetRequiredService to GetService, then null check the response and assign a new InvalidOperationException(UsefulDetailsHere) to TestBlockException if null.